pyccel/ast/numpyext.py

```diff
@@ -824,13 +824,12 @@ class NumpyNorm(PyccelInternalFunction):
 
     def __init__(self, arg, axis=None):
         super().__init__(arg, axis)
-        cmplx_precision = None
         if isinstance(arg.dtype, NativeBool):
             arg = PythonFloat(arg)
         elif not isinstance(arg.dtype, (NativeComplex, NativeReal)):
             arg = PythonFloat(arg)
```

Comment:
I am also confused by this if block: if `arg.dtype` is a `NativeBool` the second condition evaluates to `True`, so why do you need the first condition? In other words, you could just write
```python
if not isinstance(arg.dtype, (NativeComplex, NativeReal)):
    arg = PythonFloat(arg)
```
and obtain exactly the same behavior as in the code.

Reply:
you are right, for this i was believing that PythonFloat doesn’t handle bool type so this block was something like:
```python
if isinstance(arg.dtype, NativeBool):
            arg = PythonFloat(PythonInt(arg))
```
after Emily's fix, bool type is handled in PythonFloat Fortran pinter. so then that block becomes:
```python
if isinstance(arg.dtype, NativeBool):
            arg = PythonFloat(arg)
```
so, it's a useless block too, thanks.